.github/workflows/emscripten.yml

```diff
@@ -0,0 +1,59 @@
+name: Emscripten build
+
+on:
+  push:
+    branches:
+    - master
+    paths-ignore:
+    - 'android/**'
+    - 'build-data/osx/**'
+    - 'doc/**'
+    - 'doxygen_doc/**'
+    - 'lang/**'
+    - 'lgtm/**'
+    - 'tools/**'
+    - '!tools/format/**'
+    - 'utilities/**'
+  pull_request:
+    branches:
+    - master
+    paths-ignore:
+    - 'android/**'
+    - 'build-data/osx/**'
+    - 'doc/**'
+    - 'doxygen_doc/**'
+    - 'lang/**'
+    - 'lgtm/**'
+    - 'tools/**'
+    - '!tools/format/**'
+    - 'utilities/**'
+
+# We only care about the latest revision of a PR, so cancel previous instances.
+concurrency:
+  group: emscripten-build-${{ github.event.pull_request.number || github.ref_name }}
+  cancel-in-progress: true
+
+jobs:
+  build_catatclysm:
+    name: Build
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - uses: mymindstorm/setup-emsdk@v13
+
+    - name: Prepare web data
+      run: ./build-scripts/prepare-web-data.sh
+
+    - name: Build with emcc (emscripten)
+      run: ./build-scripts/build-emscripten.sh
+
+    - name: Assemble web bundle
+      run: ./build-scripts/prepare-web.sh
+
+    - name: Create artifact
```

Comment:
Since #70870 is now uploading WebAssembly builds, should we remove this step here?

Reply:
It's still handy for development—the emscripten SDK doesn't work well on macOS or Windows, so it's handy to be able to download the built artifact from CI for testing. I'm not aware of any major downsides to uploading it.

Reply:
I thought `upload-artifact` shares the total storage limit of the repository, so too many build artifact uploads could evict other data like ccache.

But I just re-checked GitHub Actions doc and it only says GitHub Actions storage usage is free for public repositories without mentioning storage limits or data eviction policy, so I'm not too sure now.